src/nwb_conversion_tools/tools/neo/neo.py

```diff
@@ -0,0 +1,602 @@
+from pathlib import Path
+import warnings
+import numpy as np
+
+import neo.io.baseio
+import pynwb
+
+from ..nwb_helpers import add_device_from_metadata
+from ...utils import OptionalFilePathType
+
+
+response_classes = dict(
+    voltage_clamp=pynwb.icephys.VoltageClampSeries,
+    current_clamp=pynwb.icephys.CurrentClampSeries,
+    izero=pynwb.icephys.IZeroClampSeries,
+)
+
+stim_classes = dict(
+    voltage_clamp=pynwb.icephys.VoltageClampStimulusSeries,
+    current_clamp=pynwb.icephys.CurrentClampStimulusSeries,
+)
+
+
+# TODO - get electrodes metadata
+def get_electrodes_metadata(neo_reader, electrodes_ids: list, block: int = 0) -> list:
+    """
+    Get electrodes metadata from Neo reader.
+
+    The typical information we look for is the information accepted by pynwb.icephys.IntracellularElectrode:
+      - name – the name of this electrode
+      - device – the device that was used to record from this electrode
+      - description – Recording description, description of electrode (e.g., whole-cell, sharp, etc)
+      - comment: Free-form text (can be from Methods)
+      - slice – Information about slice used for recording.
+      - seal – Information about seal used for recording.
+      - location – Area, layer, comments on estimation, stereotaxis coordinates (if in vivo, etc).
+      - resistance – Electrode resistance COMMENT: unit: Ohm.
+      - filtering – Electrode specific filtering.
+      - initial_access_resistance – Initial access resistance.
+
+    Parameters
+    ----------
+    neo_reader ([type]): Neo reader
+    electrodes_ids (list): List of electrodes ids.
+    block (int, optional): Block id. Defaults to 0.
+
+    Returns
+    -------
+    list: List of dictionaries containing electrodes metadata.
+    """
+    return []
+
+
+def get_number_of_electrodes(neo_reader) -> int:
+    """Get number of electrodes from Neo reader."""
+    # TODO - take in account the case with multiple streams.
+    return len(neo_reader.header["signal_channels"])
+
+
+def get_number_of_segments(neo_reader, block: int = 0) -> int:
+    """Get number of segments from Neo reader."""
+    return neo_reader.header["nb_segment"][block]
+
+
+def get_command_traces(neo_reader, block: int = 0, segment: int = 0, cmd_channel: int = 0) -> Tuple[list, str, str]:
+    """
+    Get command traces (e.g. voltage clamp command traces).
+
+    Parameters
+    ----------
+    neo_reader : neo.io.baseio
+    block : int, optional.
+        Defaults to 0.
+    segment : int, optional
+        Defaults to 0.
+    cmd_channel : int, optional
+        ABF command channel (0 to 7). Defaults to 0.
+    """
+    try:
+        traces, titles, units = neo_reader.read_raw_protocol()
+        return traces[segment][cmd_channel], titles[segment][cmd_channel], units[segment][cmd_channel]
+    except Exception as e:
+        msg = ".\n\n WARNING - get_command_traces() only works for AxonIO interface."
+        e.args = (str(e) + msg,)
+        return e
```

Comment:
Did we ever get an answer on this one? What can go wrong, why, and is there not a more specific error like `NotImplementedError`?

Reply:
also, I don't see anywhere were this function is called. @luiztauffer, what is the intended purpose of this function? How would it ideally fit into a conversion workflow?

Reply:
this is probably from an older version and its use has been replaced
the `read_raw_protocol` is specific for Icephys, and other NEO readers would not work with it. This function would unpack the content, which is basically traces, title and unit for the stimulation signals
this could be useful I guess, but right now we're not using it, so we could remove it if you prefer